sdk/src/compressed_account.rs

```diff
@@ -262,67 +157,41 @@ where
 {
     input_account: T,
     output_account: T,
-    address_seed: Option<[u8; 32]>,
-    merkle_context: PackedMerkleContext,
-    merkle_tree_root_index: u16,
-    address_merkle_context: PackedAddressMerkleContext,
+    output_compressed_account: CompressedAccount,
 }
 
 impl<T> LightMutAccount<T>
 where
     T: AnchorDeserialize + AnchorSerialize + Clone + DataHasher + Discriminator,
 {
-    pub fn try_from_slice(
-        v: &[u8],
-        merkle_context: &PackedMerkleContext,
-        merkle_tree_root_index: u16,
-        address_merkle_context: &PackedAddressMerkleContext,
-    ) -> Result<Self> {
-        let account = T::try_from_slice(v)?;
+    pub fn try_from_slice(compressed_account: &CompressedAccount) -> Result<Self> {
+        let account = T::try_from_slice(&compressed_account.data.as_ref().unwrap().data)?;
+
+        // Clean the provided compressed account from data and use it as an
+        // output account. The data of output data is going to be assigned in
+        // the `output_compressed_account` method.
+        let output_compressed_account = CompressedAccount {
+            owner: compressed_account.owner,
+            lamports: compressed_account.lamports,
+            address: compressed_account.address,
+            data: None,
+        };
 
         Ok(Self {
             input_account: account.clone(),
             output_account: account,
-            address_seed: None,
-            merkle_context: *merkle_context,
-            merkle_tree_root_index,
-            address_merkle_context: *address_merkle_context,
+            output_compressed_account,
         })
     }
 
-    pub fn set_address_seed(&mut self, address_seed: [u8; 32]) {
-        self.address_seed = Some(address_seed);
-    }
-
-    pub fn input_compressed_account(
-        &self,
-        program_id: &Pubkey,
-        remaining_accounts: &[AccountInfo],
-    ) -> Result<PackedCompressedAccountWithMerkleContext> {
-        input_compressed_account(
-            &self.input_account,
-            &self.address_seed.unwrap(),
-            program_id,
-            &self.merkle_context,
-            self.merkle_tree_root_index,
-            &self.address_merkle_context,
-            remaining_accounts,
-        )
-    }
-
-    pub fn output_compressed_account(
-        &self,
-        program_id: &Pubkey,
-        remaining_accounts: &[AccountInfo],
-    ) -> Result<OutputCompressedAccountWithPackedContext> {
-        output_compressed_account(
-            &self.output_account,
-            &self.address_seed.unwrap(),
-            program_id,
-            &self.merkle_context,
-            &self.address_merkle_context,
-            remaining_accounts,
-        )
+    pub fn output_compressed_account(&self) -> Result<OutputCompressedAccountWithPackedContext> {
+        let mut compressed_account = self.output_compressed_account.clone();
```

Comment:
why do we need to clone here?
We could also just assemble the compressed account here. (Copying data costs CU.)

Reply:
I will take care of all `.clone()` and `.unwrap()` calls before marking as ready for review.

The reason why I had to clone is that I cannot move the `output_compressed_account` field out from `self`, because this method only holds a `&self` reference. I will try to figure something out (either `self` as a receiver or assembling the compressed account here as you suggest).